File: StoryDomain/getUserStoriesService/Controller/getUserStoriesController.go
package controller

import (
    "encoding/json"  
    "context"
    "log"
    "net/http"
    "time"

    "GetUserStoriesService/database"
    "GetUserStoriesService/Models"
    "GetUserStoriesService/middleware"

    "go.mongodb.org/mongo-driver/bson"
    "go.mongodb.org/mongo-driver/bson/primitive"
)


func GetUserStories(w http.ResponseWriter, r *http.Request) {
    
    user, ok := r.Context().Value("user").(*middleware.Claims)
    if !ok {
        http.Error(w, "NoAccess", http.StatusForbidden)
        return
    }

   
    userID, err := primitive.ObjectIDFromHex(user.Sub)
    if err != nil {
        log.Println("Error converting user ID to ObjectID:", err)
        http.Error(w, "Invalid user ID", http.StatusInternalServerError)
        return
    }

   
    client, err := database.GetMongoClient()
    if err != nil {
        http.Error(w, "Database connection error", http.StatusInternalServerError)
        return
    }

    
    userFriendCollection := client.Database("social").Collection("user_friends")
    storyCollection := client.Database("social").Collection("stories")

    var friends []Models.UserFriend
    cursor, err := userFriendCollection.Find(context.TODO(), bson.M{"user_origin": userID})
    if err != nil {
        log.Println("Error fetching user friends:", err)
        http.Error(w, "Error fetching user friends", http.StatusInternalServerError)
        return
    }
    defer cursor.Close(context.TODO())

    if err = cursor.All(context.TODO(), &friends); err != nil {
        log.Println("Error decoding user friends:", err)
        http.Error(w, "Error decoding user friends", http.StatusInternalServerError)
        return
    }

    log.Println("Friends found:", len(friends))

    var currentStories []Models.Story
    today := time.Now().Unix()

    for _, friend := range friends {
        log.Println("Checking stories for friend ID:", friend.UserFriend)
        var stories []Models.Story
        cursor, err := storyCollection.Find(context.TODO(), bson.M{"user": friend.UserFriend})
        if err != nil {
            log.Println("Error fetching stories for user:", err)
            http.Error(w, "Error fetching stories", http.StatusInternalServerError)
            return
        }
        defer cursor.Close(context.TODO())

        if err = cursor.All(context.TODO(), &stories); err != nil {
            log.Println("Error decoding stories:", err)
            http.Error(w, "Error decoding stories", http.StatusInternalServerError)
            return
        }

        log.Println("Stories found for friend:", len(stories))

        for _, story := range stories {
            createdAt := story.CreatedAt.Unix()
            exp := story.Exp.Unix()
            log.Println("Story CreatedAt:", createdAt, " Exp:", exp, " Today:", today)

            if today >= createdAt && today <= exp {
                currentStories = append(currentStories, story)
            }
        }
    }

    log.Println("Total active stories:", len(currentStories))

    w.Header().Set("Content-Type", "application/json")
    json.NewEncoder(w).Encode(map[string]interface{}{"data": currentStories})
}

File: StoryDomain/getUserStoriesService - copia/Controller/getUserStoriesController.js
var User_friend = require('../models/User_friend');
var Story = require('../models/Story');
var fs = require('fs');
var path = require('path');

const get_user_stories = async function (req, res) {
    if (req.user) {
        let friends = await User_friend.find({ user_origin: req.user.sub }).populate('user_friend');

        let today = Date.parse(new Date()) / 1000;
        var current_stories_ = [];


        for (var item of friends) {
            var current_stories = [];
            var stories = await Story.find({ user: item.user_friend._id }).populate('user');


            for (var subitem of stories) {
                var tt_created = Date.parse(subitem.createdAt) / 1000;
                var tt_exp = Date.parse(subitem.exp) / 1000;

                if (today >= tt_created && today <= tt_exp) {
                    current_stories.push(subitem);
                    current_stories_.push(subitem);
                }
            }
        }

        res.status(200).send({ data: current_stories_ });

    } else {
        res.status(403).send({ message: 'NoAccess' });
    }
}

module.exports = {
    get_user_stories
};